apps/site/src/components/pages/hub/block-data-container.tsx

```diff
@@ -53,143 +70,218 @@ export const BlockDataContainer: FunctionComponent<BlockDataContainerProps> = ({
-      if (example) {
-        if (typeof example.entityId === "string") {
-          setExampleEntityId(example.entityId);
-          if (example.properties && typeof example.properties === "object") {
-            setText(JSON.stringify(example.properties, undefined, 2));
-          }
+      let entityProperties: EntityPropertiesObject | undefined;
+
+      // Prefer the `variants` field if it exists
+      const selectedVariant = metadata.variants?.[blockVariantsTab];
+      if (selectedVariant) {
+        if (selectedVariant.examples) {
+          [entityProperties] = selectedVariant.examples;
         } else {
-          setText(JSON.stringify(example, undefined, 2));
-          setExampleEntityId(`test-entity-${metadata.name}`);
+          entityProperties = selectedVariant.properties;
         }
       } else {
-        setText("{}");
+        const examples = metadata.examples;
+        if (examples) {
+          [entityProperties] = examples;
+        }
+      }
+
+      if (entityProperties) {
+        exampleEntity.properties = entityProperties;
+        setEntity(exampleEntity);
+        return;
+      }
+    }
+
+    if (exampleGraph) {
+      const exampleEntity = exampleGraph.entities.find(
+        (graphEntity) =>
+          graphEntity.metadata.recordId.entityId ===
+            exampleGraph.blockEntityRecordId.entityId &&
+          graphEntity.metadata.recordId.editionId ===
+            exampleGraph.blockEntityRecordId.editionId,
+      );
+
+      if (exampleEntity) {
+        setEntity(exampleEntity);
+      } else {
+        throw new Error(
+          `Unable to find an entity for block ${metadata.name} in the example graph`,
+        );
       }
+    } else if (!metadata.variants && !metadata.examples) {
+      // The block hasn't supplied an example graph or a schema, so as long as there aren't any variants or examples
+      // we can pass it anything as the type should be unconstrained. We can't pass in a variant or an example as we
+      // don't know what the type should be.
 
+      /* @todo - ideally we'd generate this to ensure it stays compatible with the type */
+      const exampleThingEntity: Entity | EntityTemporal = {
+        properties: {
+          "https://blockprotocol.org/types/@blockprotocol/property-type/name/":
+            "World",
+        },
+        metadata: {
+          recordId: {
+            entityId: `test-entity-${metadata.name}`,
+            editionId: interval.start.limit,
+          },
+          entityTypeId:
+            "https://blockprotocol.org/@blockprotocol/types/entity-type/thing/v/2",
+          temporalVersioning: {
+            transactionTime: interval,
+            decisionTime: interval,
+          },
+        },
+      };
+
+      setEntity(exampleThingEntity);
+    } else {
+      throw new Error(
+        `Unable to create a new entity for block ${metadata.name} from its \`examples\` or \`variants\` as it doesn't have an associated schema`,
+      );
```

Comment:
I'm unsure of how to do error handling in this file in general, I saw some code about an "alert snackbar"..

Reply:
Well I tried re-enabling the snackbar and the frontend crashed. And I noticed the current hub doesn't cause the snackbar to actually appear.